core/common/src/main/java/alluxio/conf/Source.java

```diff
@@ -14,26 +14,83 @@
 /**
  * The source of a configuration property.
  */
-public enum Source {
-  /**
-   * The unknown source which has the lowest priority.
-   */
-  UNKNOWN,
-  /**
-   * The default property value from <code>PropertyKey</code> on compile time.
-   */
-  DEFAULT,
+public class Source implements Comparable<Source> {
```

Comment:
It is a class so that `SYSTEM_PROPERTY` can take the site properties file as an argument. I couldn't get this to work with `enum` because the site properties file is determined at runtime and enum values need to be initialized statically.

Reply:
ack.